pkg/networkplugin-syncer/handlers/ovn/handler.go

```diff
@@ -0,0 +1,112 @@
+package ovn
+
+import (
+	"errors"
+
+	goovn "github.com/ebay/go-ovn"
+	clientset "k8s.io/client-go/kubernetes"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/networkplugin-syncer/handlers/ovn/nbctl"
+)
+
+type SyncHandler struct {
+	event.HandlerBase
+	k8sClientset     clientset.Interface
+	nbctl            *nbctl.NbCtl
+	nbdb             goovn.Client
+	sbdb             goovn.Client
+	localEndpoint    *submV1.Endpoint
+	remoteEndpoints  map[string]*submV1.Endpoint
+	lastOvnGwChassis string
+}
+
+func (ovn *SyncHandler) GetName() string {
+	return "ovn-sync-handler"
+}
+
+func (ovn *SyncHandler) GetNetworkPlugin() string {
+	return "OVNKubernetes"
+}
+
+func NewSyncHandler(k8sClientset clientset.Interface) *SyncHandler {
+	return &SyncHandler{
+		remoteEndpoints: make(map[string]*submV1.Endpoint),
+		k8sClientset:    k8sClientset,
+	}
+}
+
+func (ovn *SyncHandler) Init() error {
+	if err := ovn.initClients(); err != nil {
+		return err
+	}
+
+	if err := ovn.ensureSubmarinerInfra(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ovn *SyncHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
+	ovn.localEndpoint = endpoint
+
+	return ovn.updateGatewayNode()
+}
+
+func (ovn *SyncHandler) LocalEndpointUpdated(endpoint *submV1.Endpoint) error {
+	ovn.localEndpoint = endpoint
+
+	return ovn.updateGatewayNode()
+}
+
+func (ovn *SyncHandler) LocalEndpointRemoved(endpoint *submV1.Endpoint) error {
+	if ovn.localEndpoint.Name == endpoint.Name {
+		ovn.localEndpoint = nil
+	}
+
+	return ovn.updateGatewayNode()
+}
+
+func (ovn *SyncHandler) RemoteEndpointCreated(endpoint *submV1.Endpoint) error {
+	ovn.remoteEndpoints[endpoint.Name] = endpoint
```

Comment:
I guess it's safe to assume this won't be accessed by multiple threads.

Reply:
Yes, I was under that assumption, otherwise we would need a mutex. 

Based on our slack conversation, if we started processing Node events in the future (which affected those structures) we will need a mutex.

Could be added now as a caution.